server/src/main/java/org/apache/kafka/server/ClientMetricsManager.java

```diff
@@ -493,4 +519,123 @@ public void run() {
             }
         }
     }
+
+    // Visible for testing
+    final class ClientMetricsStats {
+
+        private static final String GROUP_NAME = "ClientMetrics";
+
+        // Visible for testing
+        static final String INSTANCE_COUNT = "ClientMetricsInstanceCount";
+        static final String UNKNOWN_SUBSCRIPTION_REQUEST = "ClientMetricsUnknownSubscriptionRequest";
+        static final String THROTTLE = "ClientMetricsThrottle";
+        static final String PLUGIN_EXPORT = "ClientMetricsPluginExport";
+        static final String PLUGIN_ERROR = "ClientMetricsPluginError";
+        static final String PLUGIN_EXPORT_TIME = "ClientMetricsPluginExportTime";
+
+        // Names of sensors that are registered through client instances.
+        private final Set<String> sensorsName = ConcurrentHashMap.newKeySet();
+        // List of metric names which are not specific to a client instance. Do not require thread
+        // safe structure as it will be populated only in constructor.
+        private final List<MetricName> registeredMetricNames = new ArrayList<>();
+
+        private final Set<String> instanceMetrics = Stream.of(UNKNOWN_SUBSCRIPTION_REQUEST,
+            THROTTLE, PLUGIN_EXPORT, PLUGIN_ERROR, PLUGIN_EXPORT_TIME).collect(Collectors.toSet());
+
+        ClientMetricsStats() {
+            Measurable instanceCount = (config, now) -> clientInstanceCache.size();
+            MetricName instanceCountMetric = metrics.metricName(INSTANCE_COUNT, GROUP_NAME,
+                "The current number of client metric instances being managed by the broker");
+            metrics.addMetric(instanceCountMetric, instanceCount);
+            registeredMetricNames.add(instanceCountMetric);
+        }
+
+        public void maybeAddClientInstanceMetrics(Uuid clientInstanceId) {
+            // If one sensor of the metrics has been registered for the client instance,
+            // then all other sensors should have been registered; and vice versa.
+            if (metrics.getSensor(PLUGIN_EXPORT + "-" + clientInstanceId) != null) {
+                return;
+            }
+
+            Map<String, String> tags = Collections.singletonMap(ClientMetricsConfigs.CLIENT_INSTANCE_ID, clientInstanceId.toString());
+
+            Sensor unknownSubscriptionRequestCountSensor = metrics.sensor(
+                ClientMetricsStats.UNKNOWN_SUBSCRIPTION_REQUEST + "-" + clientInstanceId);
```

Comment:
> Total number of metrics requests GetTelemetrySubscriptionsRequests with unknown CLIENT_INSTANCE_IDs.

The above is the description of the metric in the KIP. It seems inaccurate since it should be for unknown subscriptionId instead of CLIENT_INSTANCE_ID. Also, we track it for PushTelemetryRequest 
 instead of GetTelemetrySubscriptionsRequest.

Reply:
Yes, this needs to be corrected in the KIP.